src/Raven.Server/Documents/Handlers/TransformerHandler.cs

```diff
@@ -98,18 +115,45 @@ public Task GetAll()
         }
 
         [RavenAction("/databases/*/transformers/rename", "POST")]
-        public Task Rename()
+        public async Task Rename()
         {
             var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
             var newName = GetQueryStringValueAndAssertIfSingleAndNotEmpty("newName");
+            
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            {
+                var existingTransformer = Database.TransformerStore.GetTransformer(name);
+                long index = 0;
+                using (var deleteTransformerCommand = context.ReadObject(new DynamicJsonValue
```

Comment:
This is a bit wrong, instead of sending the two command as independent operations, and thus forcing us to view them as distinct operations, send them at the same time.